pkg/core/plugins/interfaces.go

```diff
@@ -80,3 +83,10 @@ type AuthnAPIServerPlugin interface {
 	Plugin
 	NewAuthenticator(PluginContext) (authn.Authenticator, error)
 }
+
+// PolicyPlugin a plugin to add a Policy to Kuma
+type PolicyPlugin interface {
+	Plugin
+	MatchedPolicies(dataplane *core_mesh.DataplaneResource, resources xds_context.Resources) (core_xds.TypedMatchingPolicies, error)
+	Apply(rs *core_xds.ResourceSet, proxy *core_xds.Proxy) error
```

Comment:
I think you are missing `ctx xds_context.Context`

Reply:
I looked at it and at least in the case of OPA it isn't used. But looking at it it sounds like at least some of it should be passed down. Do you think everything should be passed down? There are things like secrets, CLACache, Zone, VIPDomains, ServiceTLSReadiness. All these might be too much?

Reply:
Let's assume that every policy will end up as a plugin.
We will need Context to generate Envoy config for those policies. IMHO I'd pass everything. I feel that if we pick only a couple of things now (like Mesh) we will be adding more and more stuff here and eventually just end up with `xds_context.ContextV2`